website/addons/s3/static/s3NodeConfig.js

```diff
@@ -261,16 +273,62 @@ ViewModel.prototype.openCreateBucket = function() {
 
     var isValidBucket = /^(?!.*(\.\.|-\.))[^.][a-z0-9\d.-]{2,61}[^.]$/;
 
-    bootbox.prompt('Name your new bucket', function(bucketName) {
-        if (!bucketName) {
-            return;
-        } else if (isValidBucket.exec(bucketName) == null) {
-            bootbox.confirm({
-                title: 'Invalid bucket name',
-                message: 'Sorry, that\'s not a valid bucket name. Try another name?',
-                callback: function(result) {
-                    if (result) {
-                        self.openCreateBucket();
+    // Generates html options for key-value pairs in BUCKET_LOCATION_MAP
+    function generateBucketOptions(locations) {
+        var options = '';
+        for (var location in locations) {
+            if (self.settings.bucketLocations.hasOwnProperty(location)) {
+                options = options + ['<option value="', location, '">', locations[location], '</option>', '\n'].join('');
+            }
+        }
+        return options;
+    }
+
+    bootbox.dialog({
+        title: 'Create a new bucket',
+        message:
+                '<div class="row"> ' +
+                    '<div class="col-md-12"> ' +
+                        '<form class="form-horizontal"> ' +
+                            '<div class="form-group"> ' +
+                                '<label class="col-md-4 control-label" for="bucketName">Bucket Name</label> ' +
+                                '<div class="col-md-4"> ' +
```

Comment:
This markup doesn't seem like correct Bootstrap styling. I believe you need to surround form-control elements with a form-group.